src/app.module.ts

```diff
@@ -1,9 +1,52 @@
 import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { ConfigModule } from '@nestjs/config';
+import * as Joi from '@hapi/joi';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
 
 @Module({
-  imports: [],
+  imports: [
+    ConfigModule.forRoot({
+      isGlobal: true,
+      validationSchema: Joi.object({
+        APP_PORT: Joi.number().default(3000),
+        DB_DATABASE_TEST: Joi.string().required(),
+        DB_DATABASE: Joi.string().required(),
+        DB_HOST: Joi.string().required(),
+        DB_PASSWORD: Joi.string().required(),
+        DB_PORT: Joi.number().required(),
+        DB_USERNAME: Joi.string().required(),
+        JWT_SECRET: Joi.string().required(),
+        TYPEORM_ATTEMPTS: Joi.string().default(10),
+        TYPEORM_DELAY: Joi.string().default(3000),
+        TYPEORM_SYNCHRONIZE: Joi.boolean().default(false),
+      }),
+    }),
+    TypeOrmModule.forRootAsync({
+      useFactory: () => ({
+        type: 'postgres',
+        host: process.env.TYPEORM_HOST,
+        port: parseInt(process.env.TYPEORM_PORT),
+        username: process.env.TYPEORM_USERNAME,
+        password: process.env.TYPEORM_PASSWORD,
+        database:
+          process.env.NODE_ENV != 'test'
```

Comment:
You can use the same env var(`TYPEORM_DATABASE`) and just set that var in each environment. 

For example, in your local that var should be set something like: htttp://localhost but in prod o test should be set with any ip address.   

what do you think?

Reply:
I think that it's better scope, thanks!